### PR TITLE
Fix eight warnings on Linux.

### PR DESCRIPTION
**Pull request by erg**

### CODE REVIEW

File: lib/IDE/SyntaxModel.cpp [hunk 1]
```diff
@@ -1318,6 +1315,7 @@ bool ModelASTWalker::findUrlStartingLoc(StringRef Text,
       return true;
     }
   }
+#endif
   return false;
```
Comment (gribozavr) [inline lib/IDE/SyntaxModel.cpp]: Wouldn't this cause equivalent unreachable code warnings for `return false`?

Could you do `#ifdef ... #else ... #endif` instead?

File: lib/IDE/SyntaxModel.cpp [hunk 2]
```diff
@@ -1372,7 +1368,9 @@ Optional<SyntaxNode> ModelASTWalker::parseFieldNode(StringRef Text,
   StringRef MatchStr(Match.first, Match.second - Match.first);
   auto Loc = OrigLoc.getAdvancedLoc(MatchStr.data() - OrigText.data());
   CharSourceRange Range(Loc, MatchStr.size());
-  return Optional<SyntaxNode>({ SyntaxNodeKind::DocCommentField, Range });
+  Node = Optional<SyntaxNode>({ SyntaxNodeKind::DocCommentField, Range });
+#endif
+  return Node;
```
Comment (gribozavr) [inline lib/IDE/SyntaxModel.cpp]: Ditto.

File: lib/SILAnalysis/AliasAnalysis.cpp [hunk 3]
```diff
@@ -362,6 +362,11 @@ static bool isTypedAccessOracle(SILInstruction *I) {
   }
 }
 
+
+// TODO: remove once -Wdocumentation stops warning for @in, @in_guaranteed
```
Comment (gribozavr) [inline lib/SILAnalysis/AliasAnalysis.cpp]: These should be just escaped: `\@in` etc.

File: lib/Immediate/REPL.cpp [hunk 4]
```diff
@@ -183,8 +187,6 @@ static bool appendToREPLFile(SourceFile &SF,
   return FoundAnySideEffects;
 }
 
-#if defined(__APPLE__)
-
```
Comment (gribozavr) [inline lib/Immediate/REPL.cpp]: This file LGTM.

File: stdlib/public/runtime/Casting.cpp [hunk 5]
```diff
@@ -1400,6 +1401,7 @@ static void *_dynamicCastUnknownClass(void *object,
 
   return const_cast<void*>(swift_dynamicCastUnknownClass(object, targetType));
 }
+#endif
```
Comment (gribozavr) [inline stdlib/public/runtime/Casting.cpp]: LGTM.

File: lib/IDE/SyntaxModel.cpp (hunk 1 above)
Comment (slavapestov) [inline lib/IDE/SyntaxModel.cpp]: It looks like the 'return false' is reachable with and without SWIFT_HAVE_WORKING_STD_REGEX?

Comment (gribozavr) [inline lib/IDE/SyntaxModel.cpp]: Oh, I see. Cute.

Comment (erg) [inline lib/IDE/SyntaxModel.cpp]: I patched it both ways but settled on this way because it looked cleaner. I'll change it back I guess.

**Comment by slavapestov** [comment]
I think the latest version is OK. Merging.

**Pull request merged**